cmd/ethereum/state_root.go

```diff
@@ -0,0 +1,44 @@
+package ethereum
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func GetStateRoot() {
+	client, err := ethclient.Dial("")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	headers := make(chan *types.Header)
+	sub, err := client.SubscribeNewHead(context.Background(), headers)
```

Comment:
you are not using a subscription in any meaningful way

Reply:
as I understand it, `client.SubscribeNewHead`, subscribes to notification, on the channel that is passed to it, which in this case is the headers, then we check if the subscription itself has any errors in the loop 

```
case err := <-sub.Err():
			log.Fatal(err)
```
otherwise when a value is passed to the headers channel, then we use it's hash to fetch the block

Reply:
Sure but you don't use the incoming head block notifications, do you?

Reply:
We do, here 
```
case header := <-headers:
    block, err := client.BlockByHash(context.Background(), header.Hash())
     if err != nil {
	log.Fatal(err)
	}

    fmt.Println(block.Root())
```
Where we get an incoming header, we take it's hash to be able to fetch the entire block, and from there we can get the state root. Unless you meant something else & I'm misunderstanding?

Reply:
You get all the data from subscription already that you would get from BlockByHash

Reply:
Noted, noted